sdk/batch/azbatch/models.go

```diff
@@ -0,0 +1,4027 @@
+type ContainerRegistryReference struct {
+	// The reference to the user assigned identity to use to access an Azure Container Registry instead of username and password.
+	IdentityReference *NodeIdentityReference
+
+	// The password to log into the registry server.
+	Password *string
+
+	// The registry URL. If omitted, the default is "docker.io".
+	RegistryServer *string
+
+	// The user name to log into the registry server.
+	Username *string
+}
+
+// CreateJobContent - Parameters for creating an Azure Batch Job.
+type CreateJobContent struct {
+	// REQUIRED; A string that uniquely identifies the Job within the Account. The ID can contain any combination of alphanumeric
+	// characters including hyphens and underscores, and cannot contain more than 64 characters. The ID is case-preserving and
+	// case-insensitive (that is, you may not have two IDs within an Account that differ only by case).
+	ID *string
+
+	// REQUIRED; The Pool on which the Batch service runs the Job's Tasks.
+	PoolInfo *PoolInfo
+
+	// Whether Tasks in this job can be preempted by other high priority jobs. If the value is set to True, other high priority
+	// jobs submitted to the system will take precedence and will be able requeue tasks from this job. You can update a job's
+	// allowTaskPreemption after it has been created using the update job API.
+	AllowTaskPreemption *bool
+
+	// The list of common environment variable settings. These environment variables are set for all Tasks in the Job (including
+	// the Job Manager, Job Preparation and Job Release Tasks). Individual Tasks can override an environment setting specified
+	// here by specifying the same setting name with a different value.
+	CommonEnvironmentSettings []EnvironmentSetting
+
+	// The execution constraints for the Job.
+	Constraints *JobConstraints
+
+	// The display name for the Job. The display name need not be unique and can contain any Unicode characters up to a maximum
+	// length of 1024.
+	DisplayName *string
+
+	// Details of a Job Manager Task to be launched when the Job is started. If the Job does not specify a Job Manager Task, the
+	// user must explicitly add Tasks to the Job. If the Job does specify a Job Manager Task, the Batch service creates the Job
+	// Manager Task when the Job is created, and will try to schedule the Job Manager Task before scheduling other Tasks in the
+	// Job. The Job Manager Task's typical purpose is to control and/or monitor Job execution, for example by deciding what additional
+	// Tasks to run, determining when the work is complete, etc. (However, a Job Manager Task is not restricted to these activities
+	// - it is a fully-fledged Task in the system and perform whatever actions are required for the Job.) For example, a Job Manager
+	// Task might download a file specified as a parameter, analyze the contents of that file and submit additional Tasks based
+	// on those contents.
+	JobManagerTask *JobManagerTask
```

Comment:
I'm sure this came straight from the API, but there's an inconsistency with how these fields are named.

`JobManagerTask`, and friends exist (JobPreparationTask, JobReleaseTask, etc..) but then we have NetworkConfiguration (ie, not JobNetworkConfiguration, like the others).

TBH, I prefer _not_ having the `Job` prefix since we're already in a CreateJobContent.

Reply:
Thanks for pointing this out, I created #24508 to track making these names consistently good. I want to avoid revising the API in this PR because Batch devs are currently renaming things upstream for all languages (they're actually weeding out redundant prefixes like this) and I don't want to block our first beta on that work or to create churn.

Reply:
Sounds good to me, it'd be ideal if they can rename it at the source!